### PR TITLE
Clarify requirements for running scripts

### LINKED ISSUE #9: TypeError: backend must be a string or a BaseBackend instance.
When I run the function_fitting.py & fraud_detection.py, there was a same error 
TypeError: backend must be a string or a BaseBackend instance.
When the code runs to this line
`eng, q = sf.Engine(1)`

btw, the version of tensorflow and strawberryfields in my environment are 1.3.0 and 0.13.0

### PR DESCRIPTION
**Pull request by trbromley**
This PR does the following:

- Takes requirements closer to the top in the README
- Ensures that all scripts work for Strawberry Fields version 0.10, requiring the fraud detection scripts to be changed from the v0.11+ interface to v0.10
- Adds a script that raises exceptions if the strict requirements are not met. This might seem overkill, but appears the best way for now to ensure users are working with the right configuration.
- Makes the above script run for all the experiments.

We have nearly merged in support for TF2 in SF: https://github.com/XanaduAI/strawberryfields/pull/323
Once done and a new version of SF is released, it would be ideal to carefully go through these scripts and update to the new TF2 backend.

Closes https://github.com/XanaduAI/quantum-neural-networks/issues/9.

### CODE REVIEW

File: version_check.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+if python_version > (3, 6):
+    raise SystemError("Your version of python is {}.{}. You must have Python 3.5 or 3.6 installed "
+                      "to run this script.".format(python_version.major, python_version.minor))
+
+if python_version < (3, 5):
+    raise SystemError("Your version of python is {}.{}. You must have Python 3.5 or 3.6 installed "
+                      "to run this script.".format(python_version.major, python_version.minor))
+
+if sf_version != "0.10.0":
+    raise ImportError("An incorrect version of StrawberryFields is installed. You must have "
```
Comment (co9olguy) [inline version_check.py:21]:
```suggestion
    raise ImportError("An incompatible version of StrawberryFields is installed. You must have "
```

File: version_check.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                      "StrawberryFields version 0.10 to run this script. To install the correct "
+                      "version, run:\n >>> pip install strawberryfields==0.10")
+
+if not(tf_version[0] == "1" and tf_version[1] == "3"):
+    raise ImportError("An incorrect version of TensorFlow is installed. You must have "
```
Comment (co9olguy) [inline version_check.py:26]:
```suggestion
    raise ImportError("An incompatible version of TensorFlow is installed. You must have "
```

**Review by co9olguy — APPROVED** [review]
Thanks @trbromley

File: version_check.py (hunk 1 above)
Comment (trbromley) [reply version_check.py:21]: Thanks!

File: fraud_detection/fraud_detection.py [hunk 3]
```diff
@@ -190,21 +194,18 @@ def qnn_layer(layer_number):
...
 
     for i in range(depth):
         qnn_layer(i)
 
-# create an engine
-eng = sf.Engine('tf', backend_options={"cutoff_dim": cutoff, "batch_size": batch_size})
-
 # run the engine (in batch mode)
-state = eng.run(prog, run_options={"eval": False}).state
+state = eng.run("tf", cutoff_dim=cutoff, eval=False, batch_size=batch_size)
```
Comment (josh146) [inline fraud_detection/fraud_detection.py:208]: Was the fraud detection script added later than the others?

File: fraud_detection/testing.py [hunk 4]
```diff
@@ -20,6 +20,10 @@
 import strawberryfields as sf
 from strawberryfields.ops import Dgate, BSgate, Kgate, Sgate, Rgate
 
+import sys
+sys.path.append("..")
+import version_check
```
Comment (josh146) [inline fraud_detection/testing.py:25]: How come you need to append to the path here? Does `import ..version_check` not work?

File: version_check.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+sf_version = sf.__version__
+tf_version = tf.__version__.split(".")
+
+if python_version > (3, 6):
+    raise SystemError("Your version of python is {}.{}. You must have Python 3.5 or 3.6 installed "
+                      "to run this script.".format(python_version.major, python_version.minor))
+
+if python_version < (3, 5):
+    raise SystemError("Your version of python is {}.{}. You must have Python 3.5 or 3.6 installed "
+                      "to run this script.".format(python_version.major, python_version.minor))
```
Comment (josh146) [inline version_check.py:18]: Could these be merged into a single check? `if python_version < (3, 5) or python_version > (3, 6)`.

Wait, is comparison between tuples distributive in Python? I didn't know that!

**Review by josh146 — APPROVED** [review]

File: fraud_detection/fraud_detection.py (hunk 3 above)
Comment (trbromley) [reply fraud_detection/fraud_detection.py:208]: I think you're right, I definitely remember doing something more recently on that.

File: fraud_detection/testing.py (hunk 4 above)
Comment (trbromley) [reply fraud_detection/testing.py:25]: Yes I just tried again and couldn't get it to work - I think it's the lack of an `__init__.py`

File: version_check.py (hunk 5 above)
Comment (trbromley) [reply version_check.py:18]: Oh good spot, have updated!

**Pull request merged**